.github/workflows/ci.yml

```diff
@@ -3,24 +3,27 @@
 
 name: CI
 
-on:  
+on:
   pull_request:
-    branches: [ main ]
+    branches: [main]
 
 jobs:
   build:
-
     runs-on: ubuntu-latest
 
     strategy:
       matrix:
         node-version: [14.17.x, 16.x]
 
     steps:
-    - uses: actions/checkout@v2
-    - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v2
-      with:
-        node-version: ${{ matrix.node-version }}
-        cache: ${{ !env.ACT && 'npm' || '' }}
-    - run: npm test
+      - name: Checkout 🛎️
+        uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }} 🚧
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: ${{ !env.ACT && 'yarn' || '' }}
+      - name: Setup Yarn 🚧
+        run: ${{ env.ACT && 'npm install -g yarn' || '' }}
```

Comment:
No need to install Yarn as it is already installed inside the image we use => https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md#package-management

Reply:
I didn't realize virtual env comes with yarn by default. This is a perfect use-case for `if` though — that's the step that shouldn't be even invoked outside of `act`